src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -11,13 +11,27 @@
 using AutoRest.CSharp.Input.Source;
 using AutoRest.CSharp.Output.Models.Shared;
 using AutoRest.CSharp.Output.Models.Types;
+using Microsoft.CodeAnalysis;
 
 namespace AutoRest.CSharp.Generation.Types
 {
     internal class CSharpType
     {
         private readonly TypeProvider? _implementation;
```

Comment:
shall we change the naming to be more self-explaining?
```csharp
private readonly TypeProvider? _typeProvider;
private readonly Type? _frameworkType;
```
So that, we know the first one is for type created by the generator, the second one is the framework type.